### DISCUSSION TITLE
Setting end node slack to zero

### DISCUSSION THREAD

**Discussion by ronenbh1** [google/or-tools#2915] [Routing (and legacy CP) questions]
Hello,
How can I set the slack of the end node of each vehicle to zero?
```dimension.SlackVar(routing.End(vehicle_id)).SetMax(0)``` terminate the process with some error code.
Thanks,
Ronen

**Comment by Mizux** [comment]
There is no slack var at the end node, it won't be use anyway since it is the **end** node of the route.

**Reply by ronenbh1** [reply]
We have a reloading logic with multiple dimensions that we reset at the end of each delivery. We do that by using slack only on those end delivery nodes. We don't want the slack at the last node because we have a constraint that validates that at least one dimension cumulates the same as the counter. We add the constraint like this:
```python
solver = routing.solver()
for vehicle_index in vehicles:
    for end_index in reloading_nodes[vehicle_index] + [routing.End(vehicle_index)]:
        cnt_checks = [counter_dimension.CumulVar(end_index) == dimension.CumulVar(end_index) for dimension in dimensions]
        var_checks = [check.Var() for check in cnt_checks]
        solver.Add(solver.Sum(var_checks) >= 1)
```
The constraint works fine except at the last node because the slack allows it.
